### DISCUSSION TITLE
出现(code=1, message=The data type of field vector doesn't match, expected: FLOAT_VECTOR, got DOUBLE)>

### DISCUSSION THREAD

**Discussion by GzRichChen** [milvus-io/milvus#27662] [Q&A and General discussion]
embeddings = outputs.last_hidden_state[0].mean(dim=0)
print("embeddings:", embeddings)
print("embeddings.len:", len(embeddings))
connections.connect(host='localhost', port='19530')
law_id = FieldSchema(name="id", dtype=DataType.INT64, is_primary=True, auto_id=True)
law_vector = FieldSchema(name="vector", dtype=DataType.FLOAT_VECTOR, dim=768)

collection_schema = CollectionSchema(fields=[law_id, law_vector], auto_id=True)
collection = Collection(name='law_key', schema=collection_schema)
#collection = Collection(name='law_key')
index_param = {
"index_type": "IVF_FLAT",
"params": {"nlist": 1024},
"metric_type": "IP"
}
collection.create_index(field_name='vector', index_params=index_param)

embeddings_float = embeddings.float() 
collection.insert([embeddings_float.tolist()])

**Comment by yhmo** [comment]
```
   # data is a list of column-based data
   # since the primary key auto_id=True, there is only a list of embeddings in the "data"
    data = [
        [embeddings_float.tolist()], # a list of embeddings
    ]
    collection.insert(data)
```
